### PR TITLE
Adjust the sides of pages in mobile sizes

### LINKED ISSUE #16353: Reduce excess padding around cards in space-constrained layouts to optimize for content
Goal: remove excess padding that only gets in the way of actual content on space constrained layouts, for all pages.

---

The Overview page has a one-off that reduces card padding in mobile mode. I am pretty sure we handled this across pages at one point. Perhaps pages have been rewritten since and this feature fell away? (Overview was the first page using PF4, IIRC, so that's probably why it's local, and why I seem to remember other (pre-PF4) pages having this feature too.)

Here's what I'm talking about... it only takes effect in super-small sizes, to shave off a little excess padding... compare:

Screen Shot 2021-09-15 at 10 48 12
Screen Shot 2021-09-15 at 10 49 19

- This could be done in a more global, generic manner.
- It could drop the extra padding for all mobile, not just extreme sizes. (To be determined, but I think it might help.)
- We could consider it either as an opt-in for Cockpit or as a PF4 override. :shrug: 

Dropping the padding and stretching cards to the width for overview is @ https://github.com/cockpit-project/cockpit/blob/main/pkg/systemd/overview.scss#L196-L218

### PR DESCRIPTION
**Pull request by garrett**
Fixes #16353

### CODE REVIEW

**Comment by garrett** [comment]
I haven't checked this yet; I needed to get this branch out of the way to switch to other things, so I figured I'd share what I have as a draft PR.

It will break visual testing for mobile sizes.

**Comment by garrett** [comment]
Rebased. (A few commits could still be squashed together though.)

I think it's ready for review.

It'd be nice to land this soon. It will definitely need the pixel tests updated, but they have to run again first.

**Comment by martinpitt** [comment]
[This run](https://logs.cockpit-project.org/logs/pull-16379-20220214-165550-10c76b55-fedora-35/log.html) has the pixel updates. These look really nice, thank you! Do you want to update them yourselves, or want me to push these?

File: pkg/networkmanager/firewall.jsx [hunk 1]
```diff
@@ -943,7 +943,7 @@ export class Firewall extends React.Component {
                           <BreadcrumbItem isActive>{_("Firewall")}</BreadcrumbItem>
                       </Breadcrumb>}
                   additionalGroupedContent={
-                      <PageSection id="firewall-heading" className="firewall-heading" variant={PageSectionVariants.light}>
+                      <PageSection id="firewall-heading" variant={PageSectionVariants.light} className="firewall-heading">
```
Comment (marusak) [inline pkg/networkmanager/firewall.jsx:946]: Shouldn't this one also have `ct-pagesection-mobile`? If not, then we can revert this change?

File: pkg/systemd/overview.jsx [hunk 2]
```diff
@@ -138,10 +138,10 @@ class OverviewPage extends React.Component {
                 {this.state.rebootModal && <ShutdownModal onClose={() => this.setState({ rebootModal: false })} />}
                 {this.state.shutdownModal && <ShutdownModal shutdown onClose={() => this.setState({ shutdownModal: false })} />}
                 <Page>
-                    <PageSection variant={PageSectionVariants.light} padding={{ default: 'noPadding' }}>
+                    <PageSection variant={PageSectionVariants.light} padding={{ default: 'noPadding' }} className="ct-pagesection-mobile">
                         <SuperuserAlert />
                     </PageSection>
-                    <PageSection className='ct-overview-header' variant={PageSectionVariants.light}>
+                    <PageSection variant={PageSectionVariants.light} className='ct-overview-header'>
```
Comment (marusak) [inline pkg/systemd/overview.jsx:144]: Same?

File: pkg/networkmanager/firewall.jsx (hunk 1 above)
Comment (garrett) [reply pkg/networkmanager/firewall.jsx:946]: No, I think it needs padding as it has breadcrumbs, IIRC.

File: pkg/systemd/overview.jsx (hunk 2 above)
Comment (garrett) [reply pkg/systemd/overview.jsx:144]: Probably another one with some content inside that isn't holding a card view. (This time, the heading with system name and reboot.)

**Comment by marusak** [comment]
- [arch](https://logs.cockpit-project.org/logs/pull-16379-20220215-094728-1630e9de-arch/log.html) is busted. Being fixed in https://github.com/cockpit-project/bots/pull/2944
- [debian-stable](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-16379-20220215-115331-1630e9de-debian-stable/log.html#70) and [debian-testing](https://logs.cockpit-project.org/logs/pull-16379-20220215-115328-1630e9de-debian-testing/log.html#70) are unhappy with firewall. Being fixed in https://github.com/cockpit-project/cockpit/pull/17004
